opm/simulators/wells/MultisegmentWellEval.cpp

```diff
@@ -1554,6 +1554,18 @@ assembleICDPressureEq(const int seg,
     // top segment can not be a spiral ICD device
     assert(seg != 0);
 
+    if(this->segmentSet()[seg].segmentType() == Segment::SegmentType::VALVE) {
+        const Valve& valve = this->segmentSet()[seg].valve();
+        if (valve.status() == Opm::ICDStatus::SHUT) { // we switch here to be a zero rate equation
```

Comment:
For what it's worth you can reduce the nesting here if you use C++17's `if` with an initialisation statement
```C++
if (const auto& segment = this->segmentSet()[seg];
    (segment.segmentType() == Segment::SegmentType::VALVE) &&
    (segment.valve().status() == Opm::ICDStatus::SHUT))
{
    // Handle shut valve
}
```

Reply:
Will do that when the result is confirmed.

Reply:
I see the reason now. I should have deleted the extra `}`.